### PR TITLE
Add paste image action for flip editor

### PR DESCRIPTION
**Pull request by midenaio**

### CODE REVIEW

File: renderer/screens/flips/components/flip-shuffle.js [hunk 1]
```diff
@@ -252,18 +255,57 @@ function NonsenseImageEditor({
...
+  const [imageClipboard, setImageClipboard] = useState(null)
+
+  const isMac = process.platform === 'darwin'
+
+  mousetrap.bind(['command+v', 'ctrl+v'], function() {
+    pasteImageFromClipboard()
+    return false
+  })
+
+  function pasteImageFromClipboard() {
```
Comment (ridenaio) [inline renderer/screens/flips/components/flip-shuffle.js]: duplicated code

File: renderer/screens/flips/components/flip-pics.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ import theme from '../../../shared/theme'
 import {convertToBase64Url} from '../utils/use-data-url'
 import {IMAGE_SEARCH_PICK, IMAGE_SEARCH_TOGGLE} from '../../../../main/channels'
 import {IconButton} from '../../../shared/components/button'
+import {getImageURLFromClipboard} from '../../../shared/utils/clipboard'
+
+const mousetrap = require('mousetrap')
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-pics.js]: you can `import mousetrap from 'mousetrap'` here

File: renderer/screens/flips/components/flip-pics.js [hunk 3]
```diff
@@ -75,6 +81,28 @@ function FlipPics({pics, onUpdateFlip}) {
...
   }, [pickedUrl, selectedIndex])
 
+  useEffect(() => {
+    if (imageClipboard) {
+      onUpdateFlip([
+        ...pics.slice(0, selectedIndex),
+        imageClipboard,
+        ...pics.slice(selectedIndex + 1),
+      ])
+      setImageClipboard(null)
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-pics.js]: am not sure we need this or probably i miss smth

File: renderer/screens/flips/components/flip-pics.js [hunk 4]
```diff
@@ -14,6 +14,11 @@ import theme from '../../../shared/theme'
 import {convertToBase64Url} from '../utils/use-data-url'
 import {IMAGE_SEARCH_PICK, IMAGE_SEARCH_TOGGLE} from '../../../../main/channels'
 import {IconButton} from '../../../shared/components/button'
+import {getImageURLFromClipboard} from '../../../shared/utils/clipboard'
+
+const mousetrap = require('mousetrap')
+
+const isMac = process.platform === 'darwin'
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-pics.js]: this won't work. you don't have access to electron process here. need to pass `global.isMac` through the `preload.js`, like that

Comment (aidenaio) [reply renderer/screens/flips/components/flip-pics.js]: image
here from mac ^

Comment (aidenaio) [reply renderer/screens/flips/components/flip-pics.js]: or just _Cmd/Ctrl+V_ 🙂

File: renderer/screens/flips/components/flip-pics.js [hunk 5]
```diff
@@ -176,6 +204,18 @@ function FlipPics({pics, onUpdateFlip}) {
             <small> (150kb) </small>
           </IconButton>
 
+          <Divider vertical />
+
+          <IconButton
+            tooltip=""
+            icon={<FiCopy />}
+            onClick={() => {
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-pics.js]: _optional_
no need for `{}`, just `() => pasteImageFromClipboard()` is fine

File: renderer/screens/flips/components/flip-shuffle.js [hunk 6]
```diff
@@ -410,6 +436,16 @@ function NonsenseImageEditor({
             <small> (150kb) </small>
           </IconButton>
 
+          <IconButton
+            tooltip=""
+            icon={<FiCopy />}
+            onClick={() => {
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-shuffle.js]: same about `{}`

File: renderer/screens/flips/components/flip-shuffle.js [hunk 7]
```diff
@@ -252,18 +256,40 @@ function NonsenseImageEditor({
...
+  const isMac = process.platform === 'darwin'
+
+  mousetrap.bind(['command+v', 'ctrl+v'], function() {
+    pasteImageFromClipboard()
+    return false
+  })
+
+  function pasteImageFromClipboard() {
+    const url = getImageURLFromClipboard()
+    if (url) setImageClipboard(url)
```
Comment (aidenaio) [inline renderer/screens/flips/components/flip-shuffle.js]: better wrap `if` clause in `{}` here

Comment (aidenaio) [reply renderer/screens/flips/components/flip-shuffle.js]:
```
if (url) {
  setImageClipboard(url)
}
```

Comment (aidenaio) [reply renderer/screens/flips/components/flip-shuffle.js]: we try to keep curly braces `{}` in conditional statements as it's more visible and straight

**Pull request merged**